ivadomed/loader/loader.py

```diff
@@ -424,6 +424,83 @@ def get_pair_slice(self, slice_index, gt_type="segmentation"):
 
         return dreturn
 
+    def read_file(self, filename):
+        """Read file according to file type.
+        Args:
+            filename (str): Subject filename.
+        """
+        if "nii" in self.extension:
+            # Returns 'nibabel.nifti1.Nifti1Image' object
+            # For 'nii' and 'nii.gz' extentions
+            return nib.load(filename)
+        elif self.extension == "png":
+            # Returns data from file as a 3D numpy array
+            # Behavior for grayscale png only, behavior TBD for RGB or RBGA png
+            return np.expand_dims(imageio.imread(filename, as_gray=True), axis=-1)
+        else:
+            raise RuntimeError('Input file type not supported')
+
+    def apply_canonical(self, data):
+        """Apply nibabel as_closest_canonical function to nifti data only.
+        Args:
+            data ('nibabel.nifti1.Nifti1Image' object or 'ndarray'):
+                for nifti or png file respectively.
+        """
+        if "nii" in self.extension:
+            # Returns 'nibabel.nifti1.Nifti1Image' object
+            return nib.as_closest_canonical(data)
+        if self.extension == "png":
+            # Returns data as is in numpy array
+            return data
```

Comment:
Here, we have two `if` statements. What about the `if .. elif .. else raise` pattern like was done in `read_file`?

Reply:
Also, two `if` statements are used. What about the `if .. elif .. else raise` pattern like was done in `read_file`?